specification/src/chapter3.adoc

```diff
@@ -1,45 +1,313 @@
 [[chapter3]]
 
 == RISC-V security building blocks
-=== Root of trust
-RTM for measurement, reporting and updates
-Device Lifecycle management
-Layered attestation
-Authorized Debug
+
+This chapter outlines brief descriptions of RISC-V security building blocks discussed in this specification, together with general guidelines and links to technical specifications.
+
+See also the reference use cases chapter of this specification for common examples of how RISC-V security building blocks can be combined. 
 
 === Isolation
 
-=== Runtime integrity
+Isolation enable separation of software components executing on a Hart, as well as device assignment. Risc-V enables:
+
+* Privilege based isolation
+* Physical memory access control (hart and device initiated accesses)
+* Virtual memory management (hart and device virtualization)
+* Hypervisor extension 
+* Supervisor domains 
+
+==== Privilege levels
+
+https://github.com/riscv/riscv-isa-manual/releases/tag/Priv-v1.12[Privileged ISA]
+
+Standard privilege levels - Machine mode (M), Supervisor mode (S), and User mode (U) - enable separation of more privileged software from less privileged software.
+
+==== Hypervisor extension
+
+https://github.com/riscv/riscv-isa-manual/releases/tag/Priv-v1.12[Privileged ISA]
+
+Hypervisor extension supports standard supervisor level hypervisors. It extends S mode into Hypervisor supervisor mode (HS), and a virtual supervisor mode (VS) for guests. It also extends U mode into standard user mode (U) and virtual user mode (VU).
+
+Isolation of guests is typically enforced using MMU and virtual memory management.  Memory translation is in effect in VS and VU modes, but not in standard U mode. 
+
+Alternatively sPMP can be used instead of MMU to support static partition hypervisors, for example on systems with hard and deterministic real time requirements. 
```

Comment:
```suggestion
Alternatively sPMP can be used instead of MMU to support static partition hypervisors, for example on systems with hard and deterministic real time requirements [Note -The sPMP for Hypervisor extension has not been specified yet.] . 
```